### PR TITLE
Enable passing a selected license from options

### PR DESCRIPTION
**Pull request by Razzeee**
This is a way to get around https://github.com/jozefizso/generator-license/issues/37

I've tested this successfully with an old version of the generator.

Ported the changes to the new version, should work fine, but not tested by me. As I would need to bump all librarys. (will do that soon)

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -81,6 +81,7 @@ module.exports = class GeneratorLicense extends Generator {
         name: 'license',
         message: 'Which license do you want to use?',
         default: this.options.defaultLicense,
+        when: this.options.license == null,
```
Comment (SBoudrias) [inline app/index.js]: Let's validate the license that's passed in is a valid one.

Comment (Razzeee) [inline app/index.js]: I've tried `when: this.options.license == null || !(this.options.license in licenses),` but it doesn't seem to work, any pointers?

Comment (SBoudrias) [inline app/index.js]: `in` operator in JavaScript only check for key match - not content.

You'll want `license.includes(this.options.license)`

Comment (Razzeee) [inline app/index.js]: Doesn't work, as I don't have the object but the value.
I've now have this, which works in all three test cases for me:
`when: this.options.license == null || licenses.find(x => x.value === this.options.license) === undefined,`

Testcases:
- Without license specified - Prompt
- Not known license specified -> Prompt
- License specified in options -> No prompt

**Comment by SBoudrias** [comment]
Would you mind adding a test?

**Comment by Razzeee** [comment]
Like this? Please let me know if that's not okay.

File: app/index.js [hunk 2]
```diff
@@ -81,6 +81,7 @@ module.exports = class GeneratorLicense extends Generator {
         name: 'license',
         message: 'Which license do you want to use?',
         default: this.options.defaultLicense,
+        when: this.options.license == null || licenses.find(x => x.value === this.options.license) === undefined,
```
Comment (SBoudrias) [inline app/index.js]: `== null` is pretty okay IMO. It's rare we really care if something is null or undefined?

Comment (ek9) [inline app/index.js]: I believe this would be sufficient and do same thing:

    when: !this.options.license

Comment (Razzeee) [inline app/index.js]: Well this is analogue to the implementation on the other prompts. If you want this changed, it should be a different PR just touching this part and aligning all the when conditions on the prompts at once.

Comment (ek9) [inline app/index.js]: A comparison like that (`licenses.find(x => x.value === this.options.license) === undefined`) is never used in this generator. All the other conditions in the prompts are ok, unless I am missing something?

Comment (SBoudrias) [inline app/index.js]: I think the comment was about the `== null` like the other prompt. For this prompt it doesn't matter, but for the other ones it's important to allow passing empty values. I'm in favor of keeping the equality checks consistent in this case.

**Comment by Razzeee** [comment]
Updated

**Comment by SBoudrias** [comment]
Thanks

**Comment by Razzeee** [comment]
Thanks for this project
